tools/profiles/custom_bnrg2.json

```diff
@@ -0,0 +1,18 @@
+{
+    "GCC_ARM": {
+        "common": ["-c", "-Wall", "-Wextra",
+                   "-Wno-unused-parameter", "-Wno-missing-field-initializers",
+                   "-fmessage-length=0", "-fno-exceptions", "-fno-builtin",
+                   "-ffunction-sections", "-fdata-sections", "-funsigned-char",
+                   "-MMD", "-fno-delete-null-pointer-checks",
+                   "-fomit-frame-pointer", "-Og", "-g3", "-DMBED_DEBUG",
+                   "-DMBED_TRAP_ERRORS_ENABLED=1"],
+        "asm": ["-x", "assembler-with-cpp"],
+        "c": ["-std=gnu99"],
+        "cxx": ["-std=gnu++98", "-fno-rtti", "-Wvla"],
+        "ld": ["-Wl,--gc-sections", "-Wl,--wrap,main", "-Wl,--wrap,_malloc_r",
+               "-Wl,--wrap,_free_r", "-Wl,--wrap,_realloc_r", "-Wl,--wrap,_memalign_r",
+               "-Wl,--wrap,_calloc_r", "-Wl,--wrap,exit", "-Wl,--wrap,atexit",
+               "-Wl,-n"]
```

Comment:
I always look for ways to avoid adding another profile when possible. Since this is for debugging, would it be overly intrusive to change the `-O0` to `-Og` in the main `debug.json`? @deepikabhavnani or @c1728p9 may be able to educate me a bit more.

Reply:
That would be a dream for many use cases (even if the real deal would be custom settings per file/folder!). 
I think @kjbracey-arm must know when it is wrong.

Reply:
I dont understand why this is being added. We provide 3 default profiles. If any of these 3 is not sufficient, issue should be raised, not adding a new one.